mediacloud/mediawords/util/web/user_agent/test_throttled.py

```diff
@@ -0,0 +1,62 @@
+"""test ThrottledUserAgent."""
+
+import time
+
+from mediawords.test.test_database import TestDatabaseWithSchemaTestCase
+from mediawords.test.http.hash_server import HashServer
+from mediawords.util.web.user_agent.throttled import ThrottledUserAgent
+from mediawords.util.web.user_agent.throttled import McThrottledUserAgentTimeoutException
+import mediawords.util.web.user_agent.throttled
+
+
+class TestThrottledUserAgent(TestDatabaseWithSchemaTestCase):
+    """test case for ThrottledUserAgent."""
+
+    def test_request(self) -> None:
```

Comment:
it looks like this should be a number of smaller unittests!

Reply:
I don't really buy into the orthodoxy of having lots of little functions testing the smallest possible things.

The big chunk of this is a flow of tests based on the timing of making a single request, doing some tests within the timeout period, waiting for the timeout period to end, and then doing some more tests.  I could write a test case to repeat that timing for each individual test, but that just seems more confusing (and slower!) to me.